src/api-gen/FunctionMarkdownBuilder.php

```diff
@@ -11,9 +11,13 @@ class FunctionMarkdownBuilder {
   private ?DocBlock $docblock;
 
   public function __construct(
-    string $file,
+    ?string $file = null,
```

Comment:
Why is this nullable `?string`? We are not really checking for the file being `null` in here? I looked at Spyc::YAMLLoad the propogation of function calls end up here: 

```
function loadFromSource ($input) {
    if (!empty($input) && strpos($input, "\n") === false && file_exists($input))
    return file($input);
    return $this->loadFromString($input);
  }
```

`$input` in this case is our `$file`, and if it is `null` it is going to try to load from a string. But we don't load from strings here right?

in other words, we would always want a valid file here, right?